src/lib/fields_derivers_graphql/fields_derivers_graphql.ml

```diff
@@ -0,0 +1,335 @@
+open Core_kernel
+open Fieldslib
+
+(*
+module Graphql_args = struct
+  module Input = struct
+    open Graphql_async
+
+    type 'input_type t = 'input_type Schema.Arg.arg_typ
+  end
+
+  module Creator = struct
+    open Graphql_async
+
+    type 'input_type t = String.Map.t
+
+    (*
+    type 'input_type t =
+      | E :
+          ('input_type, 'fields) Schema.Arg.arg_list * 'fields
```

Comment:
Unfortunately, I think we'll need some kind of mutability to tie the knot here. For example,
```ocaml
  module Creator = struct
    type t = unit
  end
  module Accumulator = struct
    type 'input_type t =
      | Init : (unit -> 'input_type) ref
      | Acc :
          ((unit -> 'input_type) -> unit) * ('input_type, 'fields) Schema.Arg.arg_list * 'fields
  end

  let init () = Accumulator.Init (ref (fun _ -> failwith "Zoinks"))

  let add_field t_field field acc =
    let ref_as_pipe = ref None in
    let creator () = Option.value_exn !ref_as_pipe in
    match acc with
    | Accumulator.Init creator_ref ->
        let set_creator creator = creator_ref := creator in
        ( creator
        , Accumulator.Acc (set_creator, [t_field], (fun field -> ref_as_pipe := Some field; Option.value_exn !creator_ref)) )
    | Accumulator.Acc (set_creator, args, fn) ->
        ( creator
        , Accumulator.Acc (set_creator, t_field :: args, (fun field -> ref_as_pipe := Some field; fn)) )

let finish (creator, acc) =
  match acc with
  | Accumulator.Init _ -> failwith "No fields were provided for this record.."
  | Accumulator.Acc (set_creator, args, coerce) ->
      set_creator creator;
      ...
```

Reply:
todo for tomorrow